src/cpp/rtps/xmlparser/XMLElementParser.cpp

```diff
@@ -1199,6 +1196,22 @@ XMLP_ret XMLParser::getXMLWriterQosPolicies(
                 return XMLP_ret::XML_ERROR;
             }
         }
+        else if (strcmp(name, OWNERSHIP) == 0)
+        {
+            //ownership
+            if (XMLP_ret::XML_OK != getXMLOwnershipQos(p_aux0, qos.m_ownership, ident))
+            {
+                return XMLP_ret::XML_ERROR;
+            }
+        }
+        else if (strcmp(name, OWNERSHIP_STRENGTH) == 0)
+        {
+            //ownership
+            if (XMLP_ret::XML_OK != getXMLOwnershipStrengthQos(p_aux0, qos.m_ownershipStrength, ident))
+            {
+                return XMLP_ret::XML_ERROR;
+            }
+        }
```

Comment:
I think we are not handling this case properly. With this implementation we may end up with WriterQos policies that have both SHARED_OWNERSHIP and Ownership strength defined. As seen here, the strength is used to determine if we are using exclusive or shared Ownership there:

 https://github.com/eProsima/Fast-DDS/blob/78473a0de3d878a71565ee5d6c8e1e01c34beb70/src/cpp/fastdds/subscriber/history/DataReaderInstance.hpp#L198
 
 Maybe we should force that if the Ownership kind has not been explicitly set to EXCLUSIVE_OWNERSHIP the Ownership strength is discarded.